### PR TITLE
Entity shower component

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-registry/issues/1 |
| Dependencies | -- |
| Decisions | Created `entity-shower` component |
| Animated GIF | imagem |

### CODE REVIEW

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            type: Array,
+            required: true
+        },
+        getEntity: {
+            type: Function,
+            required: true
+        },
+        getEntityName: {
+            type: Function,
+            required: true
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:44]: give it a default

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        };
+    },
+    computed: {
+        entityName() {
+            return this.getEntityName(this.entity);
+        },
+        _title() {
+            if (Array.isArray(this.title)) {
+                return this.entity ? this.title.concat([{ text: this.entityName }]) : null;
+            } else return this.title;
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:97]: brackets

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            entity: {}
+        };
+    },
+    computed: {
+        entityName() {
+            return this.getEntityName(this.entity);
+        },
+        _title() {
+            if (Array.isArray(this.title)) {
+                return this.entity ? this.title.concat([{ text: this.entityName }]) : null;
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:96]: no entity verification. if needed, move elsewhere (entityName or something)

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        },
+        getEntityName: {
+            type: Function,
+            required: true
+        },
+        getEntities: {
+            type: Function,
+            required: true
+        },
+        title: {
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:50]: Title + breadcrumbs.
Title: pure override
Breadcrumbs: appends entity at the end

Comment (NFSS10) [reply vue/components/ui/organisms/entity-show/entity-show.vue:50]: Title > breadcrumbs

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                this.deleteEntity ? { label: "Delete", value: "delete", event: "delete" } : null
+            ].filter(v => v);
+        }
+    },
+    created: async function() {
+        await this.loadEntity();
+    },
+    methods: {
+        async loadEntity() {
+            this.entity = await this.getEntity();
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:114]: Add loading and `v-bind:loaded="!loading"`

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        },
+        deleteEntity: {
+            type: Function,
+            default: null
+        },
+        errorNotification: {
+            type: Boolean,
+            default: true
+        },
+        errorNotificationMessage: {
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:70]: getErrorMessage

File: vue/components/ui/organisms/entity-show/entity-show.vue [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        },
+        editRoute: {
+            type: Object | String,
+            default: null
+        },
+        deleteEntity: {
+            type: Function,
+            default: null
+        },
+        errorNotification: {
```
Comment (NFSS10) [inline vue/components/ui/organisms/entity-show/entity-show.vue:66]: errorMessage

**Comment by gcandal** [comment]
@NFSS10 please double-check the changes I've done: https://github.com/ripe-tech/ripe-components-vue/pull/383/commits/f50786918ca85815fa84f7a0bc83d978d585326f

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**